job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java

```diff
@@ -291,13 +299,39 @@ public void setQuietPeriod(Integer seconds) throws IOException {
     }
 
     @Override public CauseOfBlockage getCauseOfBlockage() {
-        // TODO is there any legitimate reason to block a flow? maybe !isConcurrentBuild()?
+        if (isLogUpdated() && !isConcurrentBuild()) {
+            return new BecauseOfBuildInProgress(getLastBuild());
+        }
         return null;
     }
+    // TODO copied from AbstractProject
+    public static class BecauseOfBuildInProgress extends CauseOfBlockage {
```

Comment:
I suppose the class in the core is a subject for the update. <code>@Restricted</code> may be useful here